src/simulation/ApplyLoad.cpp

```diff
@@ -205,31 +210,138 @@ ApplyLoad::setupLoadContracts()
-            wasmBytes.size() + 160, salt, std::nullopt);
-        closeLedger({createTx.second});
+    auto createTx = mTxGenerator.createContractTransaction(
+        lm.getLastClosedLedgerNum() + 1, 0, contractCodeLedgerKey,
+        wasmBytes.size() + 160, salt, std::nullopt);
+    closeLedger({createTx.second});
 
-        auto instanceKey =
-            createTx.second->sorobanResources().footprint.readWrite.back();
+    releaseAssert(mTxGenerator.getApplySorobanSuccess().count() -
+                      currApplySorobanSuccess ==
+                  2);
+    releaseAssert(mTxGenerator.getApplySorobanFailure().count() == 0);
+
+    auto instanceKey =
+        createTx.second->sorobanResources().footprint.readWrite.back();
 
-        TxGenerator::ContractInstance instance;
-        instance.readOnlyKeys.emplace_back(mLoadCodeKey);
-        instance.readOnlyKeys.emplace_back(instanceKey);
-        instance.contractID = instanceKey.contractData().contract;
-        mLoadInstances.emplace(kvp.first, instance);
+    mLoadInstance.readOnlyKeys.emplace_back(mLoadCodeKey);
+    mLoadInstance.readOnlyKeys.emplace_back(instanceKey);
+    mLoadInstance.contractID = instanceKey.contractData().contract;
+    mLoadInstance.contractEntriesSize =
+        footprintSize(mApp, mLoadInstance.readOnlyKeys);
+
+    auto lh = mApp.getLedgerManager().getLastClosedLedgerHeader().header;
+    auto& bl = mApp.getBucketManager().getBucketList();
+    auto const& cfg = mApp.getConfig();
+
+    uint64_t currentKey = 0;
+
+    // int const blLedgers = 1'000'000;
+    // int const blWriteFrequency = 1000;
+    // int const blLedgers = 1;
+    // int const blWriteFrequency = 1000;
+    // int const lastBatchEntries = 100;
+    // int const lastBatchThreshold = 300;
+    LedgerEntry baseLe;
+    baseLe.data.type(CONTRACT_DATA);
+    baseLe.data.contractData().contract = mLoadInstance.contractID;
+    baseLe.data.contractData().key.type(SCV_U64);
+    baseLe.data.contractData().key.u64() = 0;
+    baseLe.data.contractData().durability = ContractDataDurability::PERSISTENT;
+    baseLe.data.contractData().val.type(SCV_BYTES);
+    mDataEntrySize = xdr::xdr_size(baseLe);
+    // Add some padding to reach the configured LE size.
+    if (mDataEntrySize <
+        mApp.getConfig().APPLY_LOAD_DATA_ENTRY_SIZE_FOR_TESTING)
+    {
+        baseLe.data.contractData().val.bytes().resize(
+            mApp.getConfig().APPLY_LOAD_DATA_ENTRY_SIZE_FOR_TESTING -
+            mDataEntrySize);
+        mDataEntrySize =
+            mApp.getConfig().APPLY_LOAD_DATA_ENTRY_SIZE_FOR_TESTING;
+    }
+    else
+    {
+        CLOG_WARNING(Perf,
+                     "Apply load generated entry size is larger than "
+                     "APPLY_LOAD_DATA_ENTRY_SIZE_FOR_TESTING: {} > {}",
+                     mApp.getConfig().APPLY_LOAD_DATA_ENTRY_SIZE_FOR_TESTING,
+                     mDataEntrySize);
+    }
+
+    for (uint32_t i = 0; i < cfg.APPLY_LOAD_BL_SIMULATED_LEDGERS; ++i)
```

Comment:
Should we separate the command to populate the bucket list from `apply-load`? The data entry count will need to be communicated between the two runs, but it would allow you to copy the bucket list files and rerun with different parameters instead of waiting for bucket list generation on each run.

Reply:
Yeah, I've considered that, but didn't have time for implementation. An alternative to an additional command would be to just only generate the BL if it doesn't contain an expected entry for the maximum id (this has a small footgun if we want to shrink the BL, but I don't think that's a significant issue). This way we won't need to add any additional commands.

Reply:
I've spent some time on this, but since we're doing more ledger-based setup (initial protocol upgrades, accounts, test contract, settings), it is not as easy as just adopting the existing bucket list and would require restructuring the benchmark setup quite significantly. Thus I think this can be done as a followup later - while BL generation takes a bit of time, it's not critical for now.